### ISSUE TITLE
The documentation should explicitly state makeEwsRequestAsync require an on premise exchange server.

### ISSUE THREAD

**Issue by rjpruitt16** [OfficeDev/office-js-docs-pr#1778]
I want to start by saying how disappoint I am in the documentation. This has been long process to discover something that should be explicitly stated at the beginning.

If I am wrong, please let me know. 

In my attempt to create an outlook add-in, I became stuck with using the function makeEwsRequestAsync [.](https://github.com/OfficeDev/generator-office/issues/527). After many unanswered git issues and stackoverflow questions, I read the function again to realize that the administrator of the exchange server must invoke an cmdlet command. There should be a method to write the add-in without needing to buy an exchange server, but I decided to play along. I used exchange online to create a new exchange server. I entered my credit card (there only exists a 1 year plan).

When I invoke the cmdlet command to set oauthentication to true. I received a message saying the command does not exist. After finally finding  the documentation on the command, I learned that this command can only be run [on-premise exchange server.](https://docs.microsoft.com/en-us/powershell/module/exchange/client-access-servers/set-webservicesvirtualdirectory?view=exchange-ps)

I will admit that the responsibility was on me to read the documentation. Still, I feel that the documentation fail to put the necessary requirements to invoke the function at the beginning. There is no connecting link from the first mention of the command [makeEwsRequestAsync](https://docs.microsoft.com/en-us/office/dev/add-ins/outlook/apis) and the cmdlet command.   The makeEwsRequestAsync and  It should have said "IMPORTANT: makeEwsRequestAsync requires you have on premise exchange server, and to setup an on premises exchange server for testing go here."

Please let me know if there is another way for me to invoke ews command because it seems I have to buy a computer to use the exchange server on premis?

**Comment by lindalu-MSFT** [comment]
@AlexJerabek Issue transferred!

**Comment by oberoiankush** [comment]
@rjpruitt16 Could you please elaborate on the issues you faced when you say you were stuck with using the makeEwsRequest function? Were there any errors being returned? What client were you testing the add-in on (outlook for Windows/MAC/OutlookWebApp)? 

Exchange server is a requirement for Outlook add-ins to work. Here's a link to the document that outlines the pre-requisites for Outlook add-ins: [Outlook add-in requirements](https://docs.microsoft.com/en-us/office/dev/add-ins/outlook/add-in-requirements).

For reference I'm also adding a link that outlines steps to create and run your first add-in :[Build your first Outlook add-in](https://docs.microsoft.com/en-us/office/dev/add-ins/quickstarts/outlook-quickstart?tabs=yeomangenerator).

Hope this helps.

**Comment by rjpruitt16** [comment]
If you want to read more about my issue, you can find [https://stackoverflow.com/questions/61451273/do-you-need-an-on-premise-exchange-server-to-call-makeewsrequestasync](url) on stackoverflow. The problem I have with documentation is that it does not mention explicitly any where that you need a on premise exchange server to use almost every api useful for building an add in. It something that is not mention in the tutorial you posted nor in the requirements for the makeEwsRequestAsync. The latter notes you must set oauth to be true, but only after finding the command to set oauth, one come to the realization that you need an on premise exchange server. I spent so much time trying to create an add-in only to come to this realization that it is impossible without an 800 exchange server license. I was hoping you could put at the front of  [makeEwsRequestAsync](https://docs.microsoft.com/en-us/javascript/api/outlook/office.mailbox?view=outlook-js-preview#makeewsrequestasync-data--callback--usercontext-) and [here](https://docs.microsoft.com/en-us/office/dev/add-ins/outlook/web-services#authentication-and-permission-considerations-for-makeewsrequestasync) to explicitly state "In order to set oauth to true, you need an on premise exchange server. If you do not have one, you can use x to verify your plugin will work." I believe X does not exist, but if you know of a workaround, please let us know so no one will believe they have to spend 800 dollars on an exchange license. I tried using an online exchange server but it did not work.


https://docs.microsoft.com/en-us/javascript/api/outlook/office.mailbox?view=outlook-js-preview#makeewsrequestasync-data--callback--usercontext-

**Comment by rjpruitt16** [comment]
From my understanding, the code should work with an cloud exchange server, but there seems to be an bug that does not allow for it right now. So a better solution would be to write in makeEwsRequestAsync that it should work with online exchange servers by default. However, there is a bug currently.

**Comment by ElizabethSamuel-MSFT** [comment]
@rjpruitt16 It seems like you've opened issue https://github.com/OfficeDev/office-js/issues/1139 to report this possible bug. Since there doesn't appear to be a doc need here, I'll close this issue.

If you hit any other issue, as appropriate do comment in this issue thread, or open a new issue in the [office-js](https://github.com/OfficeDev/office-js/issues) product repo or here in the docs repo.

Thanks.
